sql/core/src/test/scala/org/apache/spark/sql/execution/debug/DebuggingSuite.scala

```diff
@@ -48,4 +50,25 @@ class DebuggingSuite extends SparkFunSuite with SharedSQLContext {
     assert(res.forall{ case (subtree, code) =>
       subtree.contains("Range") && code.contains("Object[]")})
   }
+
+  test("debug() or broadcast and columnar") {
+    val rightDF = spark.range(10)
+    val leftDF = spark.range(10)
+    val joinedDF = leftDF.join(rightDF, leftDF("id") === rightDF("id"))
+    Try {
```

Comment:
How about checking the actual output like this?
```
    assert(joinedDF.queryExecution.sparkPlan.collect { case _: BroadcastHashJoinExec => true }.nonEmpty)
    val output = new java.io.ByteArrayOutputStream()
    Console.withOut(output) {
      joinedDF.debug()
    }
    assert(output.toString.contains("BroadcastHashJoin"))
```

Reply:
Thanks for the comment. I changed test-cases added to compare expected result to actual one. But I think it's good to handle exceptions for better error message when test-cases fail.
To do so, it's easy for us to identify which test-case fails (line number appears on the top of error message).

```
[info] - SPARK-28537: DebugExec cannot debug broadcast or columnar related queries *** FAILED *** (89 milliseconds)
[info]   debug() for broadcast failed with exception (DebuggingSuite.scala:76)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:528)
[info]   at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:527)
[info]   at org.scalatest.FunSuite.newAssertionFailedException(FunSuite.scala:1560)
```